docs/docs/mdx/writing-pages.md

```diff
@@ -0,0 +1,126 @@
+---
+title: Writing pages in MDX
+---
+
+After [installing](/docs/mdx/getting-started) the plugin, MDX files
+written in `src/pages` will turn into pages. This happens because
+`gatsby-mdx` looks for MDX files and automatically transpiles them
+so that Gatsby internals can render them.
+
+Pages are rendered at a URL that is constructed from the filesystem
+path inside `src/pages`. An MDX file at `src/pages/awesome.mdx` will
+result in a page being rendered at `mysite.com/awesome`.
+
+By default, gatsby-mdx supports frontmatter so you can define things
+like titles and paths to use in your GraphQL queries.
+
+## Frontmatter
+
+You can declare frontmatter at the beginning of your MDX document:
+
+```mdx
+---
+title: Hello, world!
+path: /hello-world
+date: 2019-01-29
+---
+
+# Hello, world!
+```
+
+Which can then be [queried with GraphQL](/docs/querying-with-graphql/):
+
+```graphql
+query {
+  allMdx {
+    edges {
+      node {
+        frontmatter {
+          path
+          title
+          date(formatString: "MMMM DD, YYYY")
+        }
+      }
+    }
+  }
+}
+```
+
+## Importing components
+
+Similarly to what you'd do in JSX, you can import and render components
+with JSX. You can also import other MDX documents.
+
+```mdx
+import { Chart } from "../components/chart"
+import FAQ from "../content/faq.mdx"
+
+# Here’s a chart
+
+The chart is rendered inside our MDX document.
+
+<Chart />
+
+<FAQ />
+```
+
+## Exports
+
+MDX supports `export` syntax as well which allows you to export metadata
+about a given document. gatsby-mdx will automatically add it to the
+GraphQL schema so you can use the exported data in your queries and
+rendering
+
+```mdx
+export const metadata = {
+  name: "World",
+  path: "/world",
+}
+
+# Hello, <span>{props.metadata.name}</span>!
+
+The heading above will say "Hello, world!".
+```
+
+### Defining a layout
+
+You can specify the layout that will wrap your component using the
+default export.
+
+```mdx
+import PurpleBorder from "../components/purple-border"
+
+# This will have a purple border
+
+export default PurpleBorde
+
+r
```

Comment:
Should this `r` be at the end of line 95?

Reply:
I don't know how to text edit 😞